API/controllers/authControllers.go

```diff
@@ -59,15 +76,24 @@ var CreateAccount = func(w http.ResponseWriter, r *http.Request) {
 		account := &models.Account{}
 		err := json.NewDecoder(r.Body).Decode(account)
 		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
 			u.Respond(w, u.Message(false, "Invalid request"))
 			return
 		}
-		resp, e := account.Create()
+
+		callerUser := getUserFromToken(w, r)
+		if callerUser == nil {
+			return
+		}
+
+		resp, e := account.Create(callerUser.Roles)
 		switch e {
 		case "internal":
 			w.WriteHeader(http.StatusInternalServerError)
-		case "clientError":
+		case "clientError", "validate":
 			w.WriteHeader(http.StatusBadRequest)
+		case "unauthorised":
+			w.WriteHeader(http.StatusForbidden)
 		case "exists":
 			w.WriteHeader(http.StatusConflict)
 		default:
```

Comment:
I think there is no point in returning string status that are then converted into http status code, we might as well return the status code directly. This was already here before the PR so I will still approve but we should consider changing this later.

Reply:
You're right about the consistency. The idea is to have two returns: one with the error message and another with error type, because you can have the same error type with different causes. I prefer to return two strings (message, type), instead of a map with the message and a string with the type as it was used before, and now we have both. But I'm not sure your proposition would adapt well to this format, because I dont think it was possible to add more than a string to go's error.

In that case, since the error types are usually a small limited set, we could create a enum type ErrType and return ErrType, string. Or create our own error structure with ErrType and string inside and return only that. Then, the controller translates ErrType to some http response code (or we put the code directly in the enum ErrType and dont mind "breaking" that separation).

Reply:
the error type is just an interface { Error() string }, so we're free to build whatever on top of this

Reply:
So you prefer the approach of creating our own error and just returning one single thing? I think it's a good idea but this will impact the entirety of controller and model (way beyond rbac), so maybe a new issue for that as well?